src/views/ClusterViz/index.js

```diff
@@ -0,0 +1,239 @@
+    setProjectClusterData: PropTypes.func.isRequired,
+    className: PropTypes.string,
+};
+
+const defaultProps = {
+    className: '',
+};
+
+const mapStateToProps = state => ({
+    activeProject: projectIdFromRouteSelector(state),
+    forcedDirectedData: forceDirectedDataSelector(state),
+    projectClusterData: projectClusterDataSelector(state),
+});
+
+const mapDispatchToProps = dispatch => ({
+    setProjectClusterData: params => dispatch(setProjectClusterDataAction(params)),
+});
+
+@BoundError(AppError)
+@connect(mapStateToProps, mapDispatchToProps)
+export default class ClusterViz extends PureComponent {
+    static propTypes = propTypes;
+    static defaultProps = defaultProps;
+
+    state = {
+        createClusterPending: true,
+        clusterDataPending: true,
+    }
+
+    componentWillMount() {
+        this.stopRequestForInitCluster();
+        this.stopRequestForClusterData();
+        this.startRequestForInitCluster();
+    }
+
+    componentWillReceiveProps(nextProps) {
+        const { activeProject } = this.props;
+        if (activeProject !== nextProps.activeProject) {
+            this.stopRequestForInitCluster();
+            this.stopRequestForClusterData();
+            this.startRequestForInitCluster(activeProject, 5);
+        }
+    }
+
+    componentWillUnmount() {
+        this.stopRequestForInitCluster();
+        this.stopRequestForClusterData();
+    }
+
+    getMaxNode = data => (
+        data.reduce((previous, current) => (
+            current.score > previous.socre ? current : previous
+        )))
+
+    clusterTableExtractor = row => row.value;
+
+    startRequestForInitCluster = () => {
+        const { activeProject } = this.props;
+        const { stopRequestForClusterData, startRequestForClusterData } = this;
+
+        const noOfClusters = 5;
+
+        const createClusterRequest = new InitProjectClusterRequest({
+            stopRequestForClusterData,
+            startRequestForClusterData,
+            setState: params => this.setState(params),
+        });
+        this.createClusterRequest = createClusterRequest.create(activeProject, noOfClusters);
+        this.createClusterRequest.start();
+    }
+
+    stopRequestForInitCluster = () => {
+        if (this.createClusterRequest) {
+            this.createClusterRequest.stop();
+        }
+    }
+
+    startRequestForClusterData = (modelId, projectId) => {
+        if (modelId) {
+            const clusterDataRequest = new ProjectClusterDataRequest({
+                setProjectClusterData: this.props.setProjectClusterData,
+                setState: params => this.setState(params),
+            });
+            this.clusterDataRequest = clusterDataRequest.create(modelId, projectId);
+            this.clusterDataRequest.start();
+        }
+    }
+
+    stopRequestForClusterData = () => {
+        if (this.clusterDataRequest) {
+            this.clusterDataRequest.stop();
+        }
+    }
+
+    headers = [
```

Comment:
Either put this in constructor or use static declaration of this is not using 'this'